examples/src/main/scala/examples/hybrid/HybridApp.scala

```diff
@@ -46,14 +46,14 @@ class HybridApp(val settingsFilename: String) extends Application {
 
   override val swaggerConfig: String = Source.fromResource("api/testApi.yaml").getLines.mkString("\n")
 
-  val miner = actorSystem.actorOf(Props(new PowMiner(nodeViewHolderRef, hybridSettings.mining)))
-  val forger = actorSystem.actorOf(Props(new PosForger(hybridSettings, nodeViewHolderRef)))
+  val miner: ActorRef = PowMinerRef(nodeViewHolderRef, hybridSettings.mining)
+  val forger: ActorRef = PosForgerRef(hybridSettings, nodeViewHolderRef)
 
-  override val localInterface: ActorRef = actorSystem.actorOf(Props(new HLocalInterface(nodeViewHolderRef, miner, forger, hybridSettings.mining)))
+  override val localInterface: ActorRef = HLocalInterfaceRef(nodeViewHolderRef, miner, forger, hybridSettings.mining)
 
   override val nodeViewSynchronizer: ActorRef =
-    actorSystem.actorOf(Props(new NodeViewSynchronizer[P, TX, HybridSyncInfo, HybridSyncInfoMessageSpec.type, PMOD, HybridHistory, SimpleBoxTransactionMemPool]
-    (networkControllerRef, nodeViewHolderRef, localInterface, HybridSyncInfoMessageSpec, settings.network, timeProvider)))
+    actorSystem.actorOf(NodeViewSynchronizerRef.props[P, TX, HybridSyncInfo, HybridSyncInfoMessageSpec.type, PMOD, HybridHistory, SimpleBoxTransactionMemPool]
```

Comment:
You could use the `apply` factory method here too, as you did in line 52.

Reply:
I actually tried to. If I write:
```
override val nodeViewSynchronizer: ActorRef =
    NodeViewSynchronizerRef.apply[P, TX, HybridSyncInfo, HybridSyncInfoMessageSpec.type, PMOD, HybridHistory, SimpleBoxTransactionMemPool]
    (networkControllerRef, nodeViewHolderRef, localInterface, HybridSyncInfoMessageSpec, settings.network, timeProvider)
```
then I get this error

```
[error] ...../Scorex/examples/src/main/scala/examples/hybrid/HybridApp.scala:55:34: type mismatch;
[error]  found   : [P <: scorex.core.transaction.box.proposition.Proposition, TX <: scorex.core.transaction.Transaction[P], SI <: scorex.core.consensus.SyncInfo, SIS <: scorex.core.network.message.SyncInfoMessageSpec[SI], PMOD <: scorex.core.PersistentNodeViewModifier, HR <: scorex.core.consensus.HistoryReader[PMOD,SI], MR <: scorex.core.transaction.MempoolReader[TX]](name: String, networkControllerRef: akka.actor.ActorRef, viewHolderRef: akka.actor.ActorRef, localInterfaceRef: akka.actor.ActorRef, syncInfoSpec: SIS, networkSettings: scorex.core.settings.NetworkSettings, timeProvider: scorex.core.utils.NetworkTimeProvider)(implicit system: akka.actor.ActorSystem)akka.actor.ActorRef <and> (networkControllerRef: akka.actor.ActorRef, viewHolderRef: akka.actor.ActorRef, localInterfaceRef: akka.actor.ActorRef, syncInfoSpec: SIS, networkSettings: scorex.core.settings.NetworkSettings, timeProvider: scorex.core.utils.NetworkTimeProvider)(implicit system: akka.actor.ActorSystem)akka.actor.ActorRef
[error]  required: akka.actor.ActorRef
[error]     NodeViewSynchronizerRef.apply[P, TX, HybridSyncInfo, HybridSyncInfoMessageSpec.type, PMOD, HybridHistory, SimpleBoxTransactionMemPool]
[error]                                  ^
```
If I remove type parameters:

```
  override val nodeViewSynchronizer: ActorRef =
    NodeViewSynchronizerRef.apply(networkControllerRef, nodeViewHolderRef, localInterface, HybridSyncInfoMessageSpec, settings.network, timeProvider)
```
then the error becomes:
```
[error] ..../Scorex/examples/src/main/scala/examples/hybrid/HybridApp.scala:55:29: inferred type arguments [Nothing,Nothing,Nothing,examples.hybrid.history.HybridSyncInfoMessageSpec.type,Nothing,Nothing,Nothing] do not conform to method apply's type parameter bounds [P <: scorex.core.transaction.box.proposition.Proposition,TX <: scorex.core.transaction.Transaction[P],SI <: scorex.core.consensus.SyncInfo,SIS <: scorex.core.network.message.SyncInfoMessageSpec[SI],PMOD <: scorex.core.PersistentNodeViewModifier,HR <: scorex.core.consensus.HistoryReader[PMOD,SI],MR <: scorex.core.transaction.MempoolReader[TX]]
[error]     NodeViewSynchronizerRef.apply(networkControllerRef, nodeViewHolderRef, localInterface, HybridSyncInfoMessageSpec, settings.network, timeProvider)
[error]                             ^
[error] ..../Scorex/examples/src/main/scala/examples/hybrid/HybridApp.scala:55:92: type mismatch;
[error]  found   : examples.hybrid.history.HybridSyncInfoMessageSpec.type
[error]  required: SIS
[error]     NodeViewSynchronizerRef.apply(networkControllerRef, nodeViewHolderRef, localInterface, HybridSyncInfoMessageSpec, settings.network, timeProvider)
[error] 
```

With `props` compiled fine. Any suggestions on how to solve this?

Reply:
It is hard to think of suggestions without actually trying them. So, let's leave this like this now. Once this MR is accepted, remind me again, and I will try to see what needs to be done.